internal/web/server.go

```diff
@@ -227,33 +227,28 @@ func (s *server) Start(ctx context.Context) error {
 	})
 	http.Handle("/", s.enrichContext(router))
 
-	bindAddr := fmt.Sprintf("%v:%d", s.Host, s.Port)
-
-	s.listener, err = net.Listen("tcp", bindAddr)
+	s.listener, err = net.Listen("tcp", net.JoinHostPort(s.Host, s.Port))
 	if err != nil {
-		// Checks if Port Conflict Error exists
-		if isPortConflictError(err) {
-			userInput := cliutils.YesNoPrompt(
-				"Port is already in use.\nShould glasskube use a different port? (Y/n): ", true)
-			if userInput {
-				s.listener, err = net.Listen("tcp", ":0")
+		// if the error is "address already in use", try to get the OS to assign a random free port
```

Comment:
I think this error handling should be in the command rather than the server itself. That could make the server start method a bit more straightforward, and the `cliutils.YesNoPrompt` would actually only be called on the CLI layer. But I would say that's something for another day.

Reply:
That's a fair point, but I think it might come with some more changes, since the server would have to ensure that all watchers, timers, etc are closed in case of an error (maybe related to the potential bug described above). Ideally, the server would perform any init logic between creating the listener (thus knowing that the address/port is "claimed") and starting the http server. Also, any teardown logic should probably be moved to a deferred function.